### PR TITLE
Add iota views to FEEval

### PR DESCRIPTION
**Pull request by peterrum**
- introduce the field `n_dofs_per_cell` alongside of `dofs_per_cell`
- provide `dof_indices()` and `quadrature_point_indices()` for range-based for loops

### CODE REVIEW

**Comment by kronbichler** [comment]
I must be missing something, but where do we have a field `n_dofs_per_cell` in `FEValues`? I can find https://github.com/dealii/dealii/blob/1fe7920f850c8cdc86d64c9922b51f746fc4fd96/include/deal.II/fe/fe_values.h#L2450 and the *function* https://github.com/dealii/dealii/blob/1fe7920f850c8cdc86d64c9922b51f746fc4fd96/include/deal.II/fe/fe_base.h#L589-L590 but the latter is in `FiniteElement` and not in `FEValues`. Furthermore, I would strongly object against replacing all the code examples using a *member variable* `dofs_per_cell` by another *member variable* `n_dofs_per_cell` when we have that name as a member function in `FiniteElement` but not as a member variable.

**Comment by peterrum** [comment]
> I must be missing something, but where do we have a field n_dofs_per_cell in FEValues?

You are right. I got confused by `FiniteElementData`. The inconsistent naming convention is really confusing and requires me always to check if I need to write `n_` or not (`dofs_per_cell` vs. `n_quadrature_points`) and if yes if it is a function or a public field. I will drop `n_dofs_per_cell` from this PR. Can we keep the `dof_indices` and `quadrature_point_indices` methods?

**Comment by kronbichler** [comment]
yes, the iota views are fine.

File: include/deal.II/matrix_free/fe_evaluation.h [hunk 1]
```diff
@@ -1483,16 +1483,16 @@ class FEEvaluationAccess<1, 1, Number, is_face, VectorizedArrayType>
  * for (const auto &cell : dof_handler.active_cell_iterators())
  *   {
  *     fe_eval.reinit(cell);
- *     for (unsigned int i=0; i<dofs_per_cell;
+ *     for (unsigned int i=0; i<n_dofs_per_cell;
```
[the remark this answers is not in the document]
Comment (drwells) [reply include/deal.II/matrix_free/fe_evaluation.h:1486]: or, if we keep `dofs_per_cell`, `fe_eval.dofs_per_cell`

**Comment by peterrum** [comment]
> yes, the iota views are fine.

@kronbichler I have remove everything except the iota views.

**Review by kronbichler — APPROVED** [review]

**Review by drwells — APPROVED** [review]

**Pull request merged**